### ISSUE TITLE
Jsonify makes Record<string, any> disappear

### ISSUE THREAD

**Issue by kentcdodds** [sindresorhus/type-fest#667]
Here's a [TS Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3nAUgZwgO2ADNE4BfOIqCEOAIiVQFoiUcY6BuAKG8bQAiAQxhC4AXmx5CJADzpucOGCGIANhCEATAFxwASigDG0LbPZRg+AOYAaOEPyIAfDzKveW42qFQ0J-HY4XThhUR4AvDUUADoNawAKLRiVdU0tAEpuIA):

```ts
import type { Jsonify } from "type-fest";

type Data = Jsonify<{
  payload: Record<string, any>;
}>;

declare const d: Data;
console.log(d.payload) // <-- Property 'payload' does not exist on type 'JsonifyObject<{}>'.
```

This feels like a bug to me. It's causing an issue for me in the upcoming version of Remix: https://github.com/remix-run/remix/issues/7246

My current workaround is to change the type to `unknown` instead of `any` (https://github.com/edmundhung/conform/pull/272 which will hopefully be merged soon). I think is the proper thing to do, but sometimes that's hard because you may not control the type. I think it's more correct to keep the record around on the type.

**Comment by kentcdodds** [comment]
Behavior confirmed via #668 PR workflow run: https://github.com/sindresorhus/type-fest/actions/runs/5959009025/job/16164096431?pr=668#step:5:11

```
  ✖  331:43  Argument of type JsonifyObject<{}> is not assignable to parameter of type { payload: Record<string, any>; }.
  Property payload is missing in type JsonifyObject<{}> but required in type { payload: Record<string, any>; }.  
```

**Comment by sindresorhus** [comment]
I agree. `Record<string, any>` is already passed through when it's top-level, and it makes sense to also do it when it's a property.

**Comment by MichaelDeBoey** [comment]
Same issue happens when having `Record<string, string>` or `Record<string, Date>`

**Comment by sachinraja** [comment]
I’ll take a look when I can but that type actually comes from Remix's initial jsonify implementation

**Comment by MichaelDeBoey** [comment]
@sachinraja Is there anything I can help you with to get this one fixed?

**Comment by pcattori** [comment]
Tried reproducing this. At first, looks like the issue is resolved since there aren't red squigglies, but then I noticed that `payload` gets inferred as a `number` 🤷

https://tsplay.dev/mZJO1m

<img width="535" alt="Screenshot 2023-10-03 at 12 17 33 PM" src="https://github.com/sindresorhus/type-fest/assets/1477317/9405c594-6bd2-4e20-a452-15dfab19ac2a">

I'm looking into this now

**Comment by pcattori** [comment]
Ok tracked this down to [this line](https://github.com/sindresorhus/type-fest/blob/main/source/jsonify.d.ts#L108):

```ts
T extends TypedArray ? Record<string, number> :
// ...
```

That `extends TypedArray` check triggers for `Record<string, any>`. Similar to how `IsAny` is checked at the beginning of `Jsonify`, we need to check for `Record<string, any>` before any other `extends Record<string, _>` checks (which `TypedArray` implicitly is).

Here's a minimal repro with suggested fix: https://tsplay.dev/N7VJqw
